### DISCUSSION TITLE
Install on Monterey on M1 Macmini

### DISCUSSION THREAD

**Discussion by splobsterman** [ReagentX/imessage-exporter#240] [Q&A] [answered]
Running the install with sudo throws:

     could not amend shell profile: '/Users/house/.bash_profile': could not write rcfile file: '/Users/house/.bash_profile': Permission denied (os error 13)

Can I install on an M1  Mac?
is the default shell zsh a problem?

**Comment by ReagentX** [comment]
`imessage-exporter` supports Apple Silicon and `zsh`. Please post the commands you are running to install the software.

**Reply by splobsterman** [reply]
This was the problematical line:
`curl https://sh.rustup.rs/ -sSf | sh`

**Comment by splobsterman** [comment, marked as answer]
Following up on my own issue:
I solved the install by installing cargo using MacPorts, which installed Rust and all required dependencies
`sudo port install cargo`
Then ran:
`cargo install imessage-exporter`
`source "$HOME/.cargo/env"`
and all is well
As I know less than nothing about Rust, I was just looking for a way to get imessage-exporter installed.
